### ISSUE TITLE
Panic when using read_box

### ISSUE THREAD

**Issue by frewsxcv** [mozilla/mp4parse-rust#6]
```rust
extern crate mp4parse;

use std::io::Cursor;

fn main() {
    let mut c = Cursor::new(b"\x00\x00\x00\x13\x66\x74\x79\x70\x30\x30\x30\x30\x30\x30\x30\x30".to_vec());
    let mut context = mp4parse::MediaContext::new();
    let _ = mp4parse::read_box(&mut c, &mut context);
}
```

panic discovered using [afl.rs](https://github.com/frewsxcv/afl.rs)

**Comment by kinetiknz** [comment]
Pushed a fix here: https://github.com/kinetiknz/mp4parse-rust/commit/55d38ac41c9f7d86df03b5c951ef6bee48900c54

**Comment by rillian** [comment]
Merged in 55d38ac41c9f7d86df03b5c951ef6bee48900c54. Thanks!
